pkg/config/handlerConfigurer.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+
+	pbd "istio.io/api/mixer/v1/config/descriptor"
+	pb "istio.io/mixer/pkg/config/proto"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/template"
+)
+
+type (
+	handlerConfigurer struct {
+		tmplRepo       template.Repository
+		typeChecker    expr.TypeChecker
+		attrDescFinder expr.AttributeDescriptorFinder
+	}
+	instancesByTemplate struct {
+		instancesNamesByTemplate map[string][]string
+	}
+)
+
+func (t *instancesByTemplate) insertInstance(tmplName string, instName string) {
+	instsPerTmpl, exists := t.instancesNamesByTemplate[tmplName]
```

Comment:
I will continue to think on this.. Since are references are based on strings, and we eventually want real objects, I don't have a obvious solution at this point.. Will keep this comment open.